sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileSuite.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.binaryfile
+
+import java.sql.Timestamp
+
+import com.google.common.io.{ByteStreams, Closeables}
+import org.apache.hadoop.fs.{GlobFilter, Path}
+
+import org.apache.spark.sql.{QueryTest, Row}
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.test.{SharedSQLContext, SQLTestUtils}
+
+class BinaryFileSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
+
+  private lazy val filePath = testFile("test-data/binaryfile-partitioned")
```

Comment:
Can we write test files only for testing time to avoid adding unnesseary binary files into our code base and make it easier to read? For instance, for binary,

```scala
val tempDir = Utils.createTempDir()
val testFile = new File(dir, "part-00000")
Files.write(binary, testFile)
```

For csv, text

```
df.write.[csv|text](...)
```